From 4e04fef034a660512ab18cda9e7ae8f4b002bf18 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:01:10 -0400
Subject: [infiniband] mlx5: Fix warning about cast of wr_id back to pointer on 32 bits

Message-id: <7589011ed0f5b0581a6f21800b184f63f5e66faa.1410425016.git.avadai@redhat.com>
Patchwork-id: 93757
O-Subject: [Patch RHEL7.1 V1 134/197] IB/mlx5: Fix warning about cast of wr_id back to pointer on 32 bits
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Roland Dreier <roland@purestorage.com>

We need to cast wr_id to unsigned long before casting to a pointer.
This fixes:

       drivers/infiniband/hw/mlx5/mr.c: In function 'mlx5_umr_cq_handler':
    >> drivers/infiniband/hw/mlx5/mr.c:724:13: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
          context = (struct mlx5_ib_umr_context *)wc.wr_id;

Reported-by: kbuild test robot <fengguang.wu@intel.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 6c9b5d9b00ed2b1cbd5e5d2c176bf88da7beb224)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/mlx5/mr.c b/drivers/infiniband/hw/mlx5/mr.c
index 14ee4fd..afa873b 100644
--- a/drivers/infiniband/hw/mlx5/mr.c
+++ b/drivers/infiniband/hw/mlx5/mr.c
@@ -721,7 +721,7 @@ void mlx5_umr_cq_handler(struct ib_cq *cq, void *cq_context)
   if (err == 0)
    break;
 
-  context = (struct mlx5_ib_umr_context *)wc.wr_id;
+  context = (struct mlx5_ib_umr_context *) (unsigned long) wc.wr_id;
   context->status = wc.status;
   complete(&context->done);
  }
-- 
1.7.1